### PR TITLE
feat: SG number uses regex

### PR DESCRIPTION
**Pull request by andrehadianto**
[Notion link](https://www.notion.so/sally-wallet/Change-Regex-for-TT-Token-s-SG-Contact-number-from-lib-to-formsg-40c7136bf21a4853bb123046d4ac5fca) 

This PR adds... 
- change phone number validation for SG number to use more generic regex

- [x] I've kept this PR as small as possible (~600 lines) by splitting it into PRs with manageable chunks of code
- [x] I've requested reviews from 1-2 reviewers
- [ ] I've added [accessibility IDs](https://www.notion.so/dc5f4ce910f7431c84344ac79344e9f5?v=d487366816834dd4ab8dc12e0b5928f3) to my components
- [x] I've added/updated unit/integration tests
- [ ] I've added/updated [e2e tests](https://www.notion.so/e2e-Documentation-a096d3e0bf75485e85ad692af8371ef3) for at least the happy flow
- [ ] I've added [Chinese translations for i18n setup](https://www.notion.so/Translations-Dev-Guide-8c1da838982a4f59a386ec96ac6780c8)
- [ ] I've added documentation in README/[Notion](https://www.notion.so/82b92fb1007640328dab9582c0a3694e?v=3b6ce48202cf40ad8450553799b13146)

### CODE REVIEW

File: src/utils/validatePhoneNumbers.tsx [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import { PhoneNumberUtil, PhoneNumber } from "google-libphonenumber";
 
 const DEFAULT_COUNTRY_CODE = "+65";
+const DEFAULT_SG_PHONE_NUMBER_REGEX = /^[3689][0-9]{7}$/g;
```
Comment (dalsontws) [inline src/utils/validatePhoneNumbers.tsx:4]: do we need the `/g` global match here?

File: src/utils/validatePhoneNumbers.test.tsx [hunk 2]
```diff
@@ -143,5 +143,10 @@ describe("fullPhoneNumberValidator", () => {
     expect(fullPhoneNumberValidator("+6598261749")).toBe(true);
     expect(fullPhoneNumberValidator("+6598219374")).toBe(true);
     expect(fullPhoneNumberValidator("+6580230000")).toBe(true);
```
Comment (dalsontws) [inline src/utils/validatePhoneNumbers.test.tsx:145]: should we add some of the new numbers that failed the test previously?

File: src/utils/validatePhoneNumbers.tsx [hunk 3]
```diff
@@ -38,8 +39,13 @@ export const fullPhoneNumberValidator = (fullNumber: string): boolean => {
   try {
     const phoneNumberUtil = new PhoneNumberUtil();
     const parsedNumber = phoneNumberUtil.parse(fullNumber);
+    const nationalNumber = parsedNumber.getNationalNumber(); // always return numbers without whitespace
     const regionCode = phoneNumberUtil.getRegionCodeForNumber(parsedNumber);
-    return phoneNumberUtil.isValidNumberForRegion(parsedNumber, regionCode);
+    if (regionCode === "SG") {
+      return DEFAULT_SG_PHONE_NUMBER_REGEX.test(`${nationalNumber}`);
```
Comment (ryanoolala) [inline src/utils/validatePhoneNumbers.tsx:45]: issit can just do this?

```typescript
 return DEFAULT_SG_PHONE_NUMBER_REGEX.test(nationalNumber);
```

Comment (ryanoolala) [reply src/utils/validatePhoneNumbers.tsx:45]: or this is to force it to become a string?

Comment (andrehadianto) [reply src/utils/validatePhoneNumbers.tsx:45]: to force it to become a string!

File: src/utils/validatePhoneNumbers.test.tsx (hunk 2 above)
Comment (ryanoolala) [reply src/utils/validatePhoneNumbers.test.tsx:145]: To add on, i notice the AC mentioned a few cases, don't spot the tests here

```
***Given*** contact number country code is +65
***When*** I enter more than 8 digits in the contact number input field
***Then*** I will see Wrong format error message prompt
```
Add test case for "+65912345789" 

```
***Given*** contact number country code is +65
***When*** I enter a contact number that starts with 0 or 1 or 2 or 4 or 5 or 7 
***Then*** I will see the Wrong format error message prompt```
Add test cases for "+6501234578", "+6511234578", "+6521234578", "+6541234578", "+6551234578", "+6571234578"

```***Given*** contact number country code is +65
***When*** I enter a contact number that contains a special character or space or emoji
***Then*** I will see the Wrong format error message prompt```
Add test cases for "+65😋", "+65@"

**Comment by ryanoolala** [comment]
```
***Given*** contact number country code is +65
***When*** I enter a contact number that contains 8 digits only &  starts with 6 or 8 or 9  
***Then*** I am able to tag the contact number to token without an error message prompt
```

Have raised a comment in notion to ask if starting with "3" is invalid and should throw error or not

**Review by ryanoolala — APPROVED** [review]

**Review by dalsontws — APPROVED** [review]

**Pull request merged**